sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateSuite.scala

```diff
@@ -518,7 +526,17 @@ class TransformWithStateSuite extends StateStoreMetricsTest
 
       AddData(inputData, ("b", 31)), // Add data newer than watermark for "b", not "a"
       // Watermark = 31 - 10 = 21, so "a" should be timed out as timeout timestamp for "a" is 20.
-      CheckNewAnswer(("a", -1), ("b", 31)) // State for "a" should timeout and emit -1
+      CheckNewAnswer(("a", -1), ("b", 31)), // State for "a" should timeout and emit -1
+      Execute(q => {
+        // Filter for idle progress events and then verify the custom metrics for stateful operator
+        val progData = q.recentProgress.filter(prog => prog.stateOperators.size > 0)
+        assert(progData.filter(prog =>
```

Comment:
nit: probably { prog => and use one more line for } is much clearer to split out filter condition vs assert condition.